hil/ext/switches/dellnos9.py

```diff
@@ -153,11 +147,9 @@ def _get_native_vlan(self, interface):
         Similar to _get_vlans()
         """
         response = self._get_port_info(interface)
-        if response.find('NativeVlanId:') == -1:
-            return None
-        begin = response.find('NativeVlanId:') + len('NativeVlanId:')
-        end = response.find('.', begin)
-        vlan = response[begin:end]
+        match = re.search(re.compile(r'NativeVlanId:(\d+)\.'), response)
+        vlan = re.search(re.compile(r'(\d+)'), match.group())
```

Comment:
We lost handling for the error condition where the NativeVlanId was not found. Also, the vlanid is already captured in the original regex so something like this should work.

```
match = re.search(r'NativeVlanId:(\d+)\.', response)
if match is not None:
    vlan = match.group(1)
else:
    return None
```

Reply:
oh, I totally forgot about that haha. Good catch.

And, I was doing `match.group()[1]` which was just returning 'a'.